### PR TITLE
Export IIDM : Network iidm exported to base & extensions files

### PR DESCRIPTION
**Pull request by ChamseddineBhd**

### CODE REVIEW

File: iidm/iidm-xml-converter/src/test/resources/multiple-extensions-ext.xiidm [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<iidm:network xmlns:iidm="http://www.itesla_project.eu/schema/iidm/1_0" id="test" caseDate="2017-11-17T12:00:00.000+01:00" forecastDistance="0" sourceFormat="test" xmlns:bar="http://www.itesla_project.eu/schema/iidm/ext/loadbar/1_0" xmlns:foo="http://www.itesla_project.eu/schema/iidm/ext/loadfoo/1_0">
+    <iidm:extension id="LOAD">
+        <foo:loadFoo/>
+        <bar:loadBar/>
+    </iidm:extension>
+</iidm:network>
```
Comment (MioRtia) [inline iidm/iidm-xml-converter/src/test/resources/multiple-extensions-ext.xiidm]: add an empty line

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java [hunk 2]
```diff
@@ -97,22 +99,32 @@ public void export(Network network, Properties parameters, DataSource dataSource
...
+
         }
 
         try {
             long startTime = System.currentTimeMillis();
+            Anonymizer anonymizer;
+            try (OutputStream osb = dataSource.newOutputStream(null, "xiidm", false);
+                 BufferedOutputStream bos = new BufferedOutputStream(osb)) {
+
+                if (parameters != null &&  Boolean.parseBoolean(parameters.getProperty(SEPARATE_BASE_EXTENSIONS_PROPERTY, Boolean.FALSE.toString()))) {
```
Comment (MioRtia) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java]: don't check ```parameters```, check ```options``` directly

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXmlWriterContext.java [hunk 3]
```diff
@@ -33,7 +34,16 @@
         this.writer = writer;
         this.options = options;
         this.filter = filter;
+        this.extensionsWriter = null;
+        this.exportedEquipments = new HashSet<>();
+    }
```
Comment (MioRtia) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXmlWriterContext.java]: delegate to the ```NetworkXmlWriterContext(Anonymizer, XMLStreamWriter, XMLStreamWriter, ExportOptions, BusFilter)``` directly

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java [hunk 4]
```diff
@@ -97,22 +99,32 @@ public void export(Network network, Properties parameters, DataSource dataSource
...
+                    try (OutputStream ose = dataSource.newOutputStream("ext", "xiidm", false);
```
Comment (MioRtia) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java]: Same remark than with ```extensionWriter``` : should it be created here ?

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java [hunk 5]
```diff
@@ -73,6 +73,8 @@
 
     public static final String THROW_EXCEPTION_IF_EXTENSION_NOT_FOUND = "iidm.export.xml.throw-exception-if-extension-not-found";
 
+    public static final String SEPARATE_BASE_EXTENSIONS_PROPERTY = "iidm.export.base-and-extensions-separately";
```
Comment (MioRtia) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java]: name ```iidm.export.xml.base-and-extensions-separately```

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java (hunk 2 above)
Comment (ChamseddineBhd) [reply iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java]: Alright

**Pull request closed without merging**